metrics/metrics.go

```diff
@@ -165,3 +183,46 @@ func (m *Metrics) registerNodeMetrics() {
 	prometheus.Register(m.nodeMetrics.DiscardedEvents)
 	prometheus.Register(m.nodeMetrics.BufferFullEvents)
 }
+
+func (m *Metrics) registerMessageMetrics() {
+	m.messageMetrics = MessageMetrics{
+		MessagesSent: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.AppMetricsPrefix,
+				Name:      "messages_sent_total",
+				Help:      "The total number of messages sent for each message type",
+			},
+			[]string{"transport", "messagetype"},
+		),
+		MessagesReceived: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.AppMetricsPrefix,
+				Name:      "messages_received_total",
+				Help:      "The total number of messages received for each message type",
+			},
+			[]string{"transport", "messagetype"},
+		),
+		MessagesSentBytes: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.AppMetricsPrefix,
+				Name:      "messages_sent_bytes_total",
+				Help:      "The total bytes of messages sent for each message type",
+			},
+			[]string{"transport", "messagetype"},
+		),
+		MessagesReceivedBytes: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: m.AppMetricsPrefix,
+				Name:      "messages_received_bytes_total",
+				Help:      "The total bytes of messages received for each message type",
+			},
+			[]string{"transport", "messagetype"},
+		),
+	}
+
+	// nolint:errcheck
+	prometheus.Register(m.messageMetrics.MessagesSent)
+	prometheus.Register(m.messageMetrics.MessagesSent)
```

Comment:
One of them is probably meant for the registration of m.messageMetrics.MessageReceived and not MessagesSent twice?

Reply:
there it is 👍